Documentation: Move "Modules" into "Configuration"

Arnei opened the pull request:
The differentiation between "modules" and "configuration" in the documentation may been useful in the past, but is not anymore today. From an admin perspective, I don't care if the thing I'm configuring is a "module" or not, and either end up looking through both "modules" and "configuration" or just using the search function.

This PR gets rid of the "modules" tab by moving all its pages to the "configuration" tab. It also takes the opportunity to consolidate information and remove/rework outdated information

Arnei committed:
Docs: Move "Modules" into "Configuration"

From an admin perspective it does not matter if they are configuring
a module or not. Therefore, this commit groups them.

Arnei committed:
Docs: Consolidate OpenSearch/Elasticsearch config

Don't need the same configuration twice or in multiple places.

Arnei committed:
Docs: Small correction on Solr usage

Arnei committed:
Docs: Sort Admin UI Configs

Arnei committed:
Docs: Consolidate video editor information

Regroups information on the video editor and deletes outdated/duplicate information.

Arnei committed:
Docs: Configuration overview

Adds some new/missing configurations to the list.
Also attempts a bit of thematic grouping. Not sure if I like where
it is going yet.

Arnei committed:
Docs: Remove editor trimming page

It just repeated what the configuration file itself also stated, so
I just put a link to it in on the relevant page. Also someone should
check if those values are actually still used by the editor

Arnei committed:
Docs: Remove waveform page

Information on the configuration was outdated. Instead put a
link to the configuration in the relevant place

Arnei committed:
Docs: Fix layout typo

Arnei committed:
Merge pull request #454 from Arnei/documentation-rework-configuration-module

Merge Configuration and Modules

Arnei committed:
Docs: Group Player Configuration

Consolidate the three player configurations into one group.
Also added a little note on why there are two player in
the overview.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into documentation-rework-halle

Arnei committed:
Docs: Fix relative links

Bunch of relative links in the docs got broken,
this should fix them.

Arnei commented:
Don't understand why tests are failing, they run fine locally.

KatrinIhler commented:
> Don't understand why tests are failing, they run fine locally.

Could it be that you're running a different version of markdownlint locally that is missing certain checks? I ran into this problem recently with ansible linting.

Arnei commented:
I did run `npm install` before `npm start`, so there shouldn't be any major version differences.

KatrinIhler commented:
Hm, then I don't know. I guess you could always try to fix the stuff the pipeline is complaining about :man_shrugging:

ziegenberg commented:
Looking at `docs/checkstyle/check-docs.sh`, it fails while running `mkdocs build` for the developer docs.

```
Building developer documentation…
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
WARNING -  A relative path to 'architecture/new-admin-ui.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A relative path to 'architecture/stream-security-config.md' is included in the 'nav' configuration, which is not found in the documentation files.
developer did not build correctly!
```

ziegenberg commented on docs/guides/developer/mkdocs.yml:
The two changes in this file let `mkdocs build` trip over its on feet.

Arnei replied:
Tfw you think you didn't touch the developer docs so you come to the conclusion that those warnings couldn't possibly be related *facepalm*. Thanks for pointing this out, did the trick.

ziegenberg replied:
Looks good now :)

Arnei committed:
Undo changes accidental changes to developer config nav

Accidentally changed some paths in the nav file for the
developer config while refactoring the admin config.
This undoes that.

gregorydlogan commented on docs/guides/admin/docs/index.md:
```diff
@@ -28,19 +28,17 @@ The Opencast Release Documentation is the official Opencast documentation for ea
     * [HTTPS Configuration](configuration/https/index.md)
     * [Workflow Configuration](configuration/workflow.md)
     * [Encoding Configuration](configuration/encoding.md)
+    * [Plugin Management](configuration/plugin-management.md)
+    * [Atom and RSS Feed](configuration/atomrss.md)
+    * [AWS S3 Distribution](configuration/awss3distribution.md)
+    * [Media Module](configuration/mediamodule.configuration.md)
+    * [Metrics (OpenMetrics, Prometheus)](configuration/metrics.md)
+    * [New Admin UI](configuration/admin-ui/new-admin-ui.md)
+    * [Stream Security](configuration/stream-security/stream-security-overview.md)
+    * [Stand-alone Video Editor](configuration/videoeditor.frontend.md)
+    * [Studio](configuration/studio.md)
+    * [Text Extraction](configuration/textextraction.md)
+    * [Video Segmentation](configuration/videosegmentation.md)
+    * [YouTube Publication](configuration/youtubepublication.md)
     * [more...](configuration/index.md)
```
I don't dislike this as much as I thought I would, but I would definitely edit the list of default config items shown to shorten it up.

What about this?

```suggestion
    * [Plugin Management](configuration/plugin-management.md)
    * [Atom and RSS Feed](configuration/atomrss.md)
    * [Metrics (OpenMetrics, Prometheus)](configuration/metrics.md)
    * [New Admin UI](configuration/admin-ui/new-admin-ui.md)
    * [Stream Security](configuration/stream-security/stream-security-overview.md)
    * [Stand-alone Video Editor](configuration/videoeditor.frontend.md)
    * [Studio](configuration/studio.md)
    * [more...](configuration/index.md)
```

(drops s3 dist, media module, text extraction, video segmentation, and youtube publication).  I could see arguments for removing other things (admin ui, feeds, metrics) too, but this at least shortens the list down to keep the initial shock of the list size down.

Arnei replied:
I have no opinion on what is and is not shown in that list (it always seemed somewhat superfluous to me), so I'll go along with your suggestion if no one disagrees.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into documentation-rework-halle

Arnei committed:
Shorten admin docs index list

To lessen the shock a long list can give some people.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into documentation-rework-halle

Arnei commented:
@gregorydlogan, think this could get approved today? I'd like to have this in OC15 (and not have to resolve conflicts everytime someone pushes a change to the docs).

gregorydlogan commented:
LGTM, merging.
